dotcom-rendering/playwright/tests/parallel-5/atom.video.e2e.spec.ts

```diff
@@ -0,0 +1,458 @@
+import { isUndefined } from '@guardian/libs';
+import type { Page } from '@playwright/test';
+import { expect, test } from '@playwright/test';
+import { expectToBeVisible, expectToNotExist } from 'playwright/lib/locators';
+import { cmpAcceptAll, cmpRejectAll } from '../../lib/cmp';
+import { waitForIsland } from '../../lib/islands';
+import { loadPage } from '../../lib/load-page';
+
+type YouTubeEmbedConfig = {
+	adsConfig: {
+		nonPersonalizedAd: boolean;
+		adTagParameters: {
+			cmpGdpr: number;
+			cmpGvcd: string;
+			cmpVcd: string;
+			cust_params: string;
+			iu: string;
+		};
+	};
+};
+
+const interceptOphanPlayEvent = ({ page, id }: { page: Page; id: string }) => {
+	return page.waitForRequest((request) => {
+		const matchUrl = request
+			.url()
+			.startsWith('http://ophan.theguardian.com/img/2?');
+		const searchParams = new URLSearchParams(request.url());
+		const videoSearchParam = searchParams.get('video');
+		const expectedVideoSearchParam = JSON.stringify({
+			id,
+			eventType: 'video:content:play',
+		});
+		return matchUrl && videoSearchParam === expectedVideoSearchParam;
+	});
+};
+
+const interceptYouTubeEmbed = ({
+	page,
+	videoId,
+	adUnit,
+	pageUrl,
+	rejectAll,
+}: {
+	page: Page;
+	videoId: string;
+	adUnit: string;
+	pageUrl: string;
+	rejectAll: boolean;
+}) => {
+	return page.waitForRequest((request) => {
+		const matchUrl = request
+			.url()
+			.startsWith(`https://www.youtube.com/embed/${videoId}?`);
+		if (!matchUrl) {
+			return false;
+		}
+		const searchParams = new URLSearchParams(request.url());
+		const rawEmbedConfig = searchParams.get('embed_config');
+		if (!rawEmbedConfig) {
+			return false;
+		}
+		const embedConfig = JSON.parse(rawEmbedConfig) as YouTubeEmbedConfig;
+		const adsConfig = embedConfig.adsConfig;
+		const adTagParameters = adsConfig.adTagParameters;
+		// cust_params is double encoded
+		const custParams = new URLSearchParams(
+			decodeURIComponent(adTagParameters.cust_params),
+		);
+		// check adunit
+		const adUnitMatch = adTagParameters.iu === adUnit;
+		// check url to check custParams is present
+		const urlCustParamMatch = custParams.get('url') === pageUrl;
+
+		// check consent related properties
+		// cmpGdpr = consentState.tcfv2.gdprApplies
+		const cmpGdprMatch = adTagParameters.cmpGdpr === 1;
+		// cmpVcd = consentState.tcfv2.tcString
+		const cmpVcdMatch = !isUndefined(adTagParameters.cmpVcd);
+		let nonPersonalizedAdMatch = false;
+		let cmpGvcdMatch = false;
+		if (rejectAll) {
+			// user has not given consent for any purpose
+			nonPersonalizedAdMatch = adsConfig.nonPersonalizedAd;
+			// cmpGvcd = consentState.tcfv2.addtlConsent
+			cmpGvcdMatch = adTagParameters.cmpGvcd === '1~';
+		} else {
+			// user has not given consent for any purpose
+			nonPersonalizedAdMatch = !adsConfig.nonPersonalizedAd;
+			// cmpGvcd = consentState.tcfv2.addtlConsent
+			cmpGvcdMatch = !isUndefined(adTagParameters.cmpGvcd);
+		}
```

Comment:
I think there’s a few leftover comments?

Reply:
Those are carried over as a pointer to where the GDPR properties we send to YouTube come from in the `ConsentState` object.